sonarts-sq-plugin/sonar-typescript-plugin/src/test/java/org/sonar/plugin/typescript/ExternalTypescriptSensorTest.java

```diff
@@ -383,4 +384,29 @@ protected void readErrors(BufferedReader errorReader) {
     }
   }
 
+  private static class TestableErrorConsumer extends ExternalProcessErrorConsumer {
+
+    private volatile Boolean running = null;
+    private int delay;
+
+    TestableErrorConsumer(int delay) {
+      this.delay = delay;
+    }
+
+    @Override
+    protected void readErrors(BufferedReader errorReader) {
+      running = true;
+      sleep(delay);
+      super.readErrors(errorReader);
+      running = false;
+    }
+  }
+
+  private static void sleep(int millis) {
+    try {
+      Thread.sleep(millis);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this use of "Thread.sleep()". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2925)